### PR TITLE
Add core web vital tracking in `amp-analytics`

### PR DESCRIPTION
**Pull request by micajuine-ho**

### CODE REVIEW

**Comment by sebastianbenz** [comment]
Awesome! Thanks a lot! I was wondering whether this would be better suited for a guide that we extend over time. 

Sidenote: could we also add a Google Analytics sample?

//cc @patrickkettner for thoughts

**Comment by patrickkettner** [comment]
I think having the example there is a wonderful place to start, a guide would be awesome as well, but this should be landed asap.

Only improvements I could imagine would be
- im not clear on when the preview images is shown, but it would be great to highlight the phrase Core Web Vitals, or use one of the [common images associated](https://lh5.googleusercontent.com/HlrpevA1hZKx35h2SQdwOBdCO4FOC0YOqie9eMTiGDZx5MdSVTxY2VwPwdL58Pi68cuuG0ooeRTs3RJQEfU5woNRpgq1ZLV4SrWkzHIOH4kTnLS32i62Qf7UibEcz2xm8Gb4nT_e) with the topic
- either inline comments or links to places explaining exactly what the schema and syntax of the config object is

File: examples/source/advertising-analytics/Tracking_Core_Web_Vitals.html [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+hidePreview: 'true'
+teaserImage: '/static/samples/img/teaser/analytics-pages.png'
+author: micajuineho
+
+--->
+
+<!--
+  ## Introduction
+
+  [Core Web Vital](https://blog.amp.dev/2020/05/06/amp-web-vitals-a-better-web-together/) metrics are available to developers for their AMP pages through `amp-analytics`. Variable substitution can be used to add these metrics in to any outgoing requests mader by `amp-analytics`.
```
Comment (CrystalOnScript) [inline examples/source/advertising-analytics/Tracking_Core_Web_Vitals.html:12]:
```suggestion
 AMP developers can measure [Core Web Vital](https://blog.amp.dev/2020/05/06/amp-web-vitals-a-better-web-together/) metrics through the `amp-analytics` component. Use variable substitution to add these metrics to any outgoing requests made by `amp-analytics`.
```

File: examples/source/advertising-analytics/Tracking_Core_Web_Vitals.html [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  If you would like to learn more about `amp-analytics`, you may find a summary [here](/content/amp-dev/documentation/guides-and-tutorials/optimize-measure/configure-analytics/analytics_basics.md).
```
Comment (CrystalOnScript) [inline examples/source/advertising-analytics/Tracking_Core_Web_Vitals.html:14]:
```suggestion
Learn more about `amp-analytics` in our guide [Analytics: the basics](/content/amp-dev/documentation/guides-and-tutorials/optimize-measure/configure-analytics/analytics_basics.md).
```

**Comment by micajuine-ho** [comment]
> Sidenote: could we also add a Google Analytics sample?

Done.

> use one of the common images associated with the topic

Done. 

> either inline comments or links to places explaining exactly what the schema and syntax of the config object is

I'm not sure what I have to do to address this comment.

**Review by CrystalOnScript — COMMENTED** [review]
Some small nits to remove passive voice and add a bit more structure. But otherwise it looking good!

File: examples/source/advertising-analytics/Tracking_Core_Web_Vitals.html [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            }
+          }
+        }
+      </script>
+    </amp-analytics>
+  </div>
+
+
+  <!--
+  ## Tracking Core Web Vitals while using `googleanalytics`
```
Comment (sebastianbenz) [inline examples/source/advertising-analytics/Tracking_Core_Web_Vitals.html:63]: Rephrased for better SEO

```suggestion
  ## Tracking Core Web Vitals in AMP using Google Analytics
```

**Review by sebastianbenz — APPROVED** [review]
Looks good to me (once the suggestions are one). Thanks!

File: examples/source/advertising-analytics/Tracking_Core_Web_Vitals.html [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
+</head>
+<body>
+
+  <!-- ## Using `amp-analytics` to track Core Web Vitals -->
+  <!--
+    You can add the core web vital macros (${largestContentfulPaint}, ${cumulativeLayoutShift}, ${firstInputDelay}) to any of your analytics requests. These macros will resolve asynchronously and can be attached to any trigger. These macros can only be accessed through `amp-analytics`.
+  -->
```
Comment (CrystalOnScript) [inline examples/source/advertising-analytics/Tracking_Core_Web_Vitals.html:35]:
```suggestion
The `amp-analytics` component provides the following macros:

* [Largest Contentful Pain](https://web.dev/lcp/): `${largestContentfulPaint}`
* [Cumulative Layout Shift](https://web.dev/cls/): `${cumulativeLayoutShift}`
* [First Input Delay](https://web.dev/fid/): `${firstInputDelay}`

These macros can attach to any triggers and resolve asynchronously. 
  -->
```

**Review by CrystalOnScript — APPROVED** [review]
Looks like one of my suggestions didn't make it into my last review - added it here! Otherwise LGTM!

File: examples/source/advertising-analytics/Tracking_Core_Web_Vitals.html [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  <!--
+  Analytics must be configured in the body. Here we use Google Analytics to track pageviews and use the `extraUrlParams` feature to append the Core Web Vitals to the request URL.
+  -->
+  <amp-analytics type="googleanalytics">
+    <script type="application/json">
+      {
+        "vars": {
+          "account": "UA-73836974-1"
+        },
```
Comment (sebastianbenz) [inline examples/source/advertising-analytics/Tracking_Core_Web_Vitals.html:79]: Afaik googleanalytics tag is deprecated

```suggestion
  <amp-analytics type="googleanalytics">
    <script type="application/json">
      {
          "vars": {
             "gtag_id": "UA-XXXXXXXXX-X",
             "config": {
               "UA-XXXXXXXXX-X": {
                  "groups": "default"
               }
             },
```

**Comment by sebastianbenz** [comment]
I've just tried this on amp.dev (see https://github.com/ampproject/amp.dev/pull/5273/), but it doesn't seem to work. The other thing I've been wondering: how can something like FID be tracked on the pageview event?

**Comment by sebastianbenz** [comment]
//cc @patrickkettner

**Comment by micajuine-ho** [comment]
> I've just tried this on amp.dev (see #5273), but it doesn't seem to work. The other thing I've been wondering: how can something like FID be tracked on the pageview event?

They do work they just take a long time to send and they only send after a user interacts with the page. 

We don't have a specialized trigger for CWV, so the idea is that we just attach these CWV macros onto any trigger. 

The macros will resolve through our performance service at some point in time, and once that occurs then the requests will be sent out.

Maybe a better config would be to send out separate requests for each CWV: 
```
{
  "triggers": {
    "pageviewCls": {
      "on": "visible",
      "request": "endpoint",
      "extraUrlParams": {
        "cls": "${cumulativeLayoutShift}"
      }
    },
    "pageviewLcp": {
      "on": "visible",
      "request": "endpoint",
      "extraUrlParams": {
        "lcp": "${largestContentfulPaint}"
      }
    },
    "pageviewFid": {
      "on": "visible",
      "request": "endpoint",
      "extraUrlParams": {
        "fid": "${firstInputDelay}"
      }
    }
  }
}
```

**Comment by patrickkettner** [comment]
> They do work they just take a long time to send and they only send after a user interacts with the page.

> We don't have a specialized trigger for CWV, so the idea is that we just attach these CWV macros onto any trigger.

Both of these things are not super obvious - where are they currently documented?

**Comment by patrickkettner** [comment]
sorry, I meant more of the documentation of the nuances, rather than the properties themselves. @CrystalOnScript where would be a good place to make note of the above things? On the docs page, a guide, or something else?

**Comment by micajuine-ho** [comment]
> sorry, I meant more of the documentation of the nuances, rather than the properties themselves. @CrystalOnScript where would be a good place to make note of the above things? On the docs page, a guide, or something else? 

Related: https://github.com/ampproject/amphtml/pull/32394

**Pull request merged**